### DISCUSSION TITLE
TRPC request stays as 'paused' and is never run?

### DISCUSSION THREAD

**Discussion by russ0133** [trpc/trpc#5190] [Q&A] [answered]
Some of my users are having their TRPC requests stay as "paused" indefinitely. I have a T3 app that uses TRPC, and I have some pretty straight forward dashboard UI.

This doesn't happen at my machine but happens on some of the user's machines, here's a sample code:

```
console.log("Fetching products!!!");
const products = api.product.getAllWithRelations.useQuery({
    includeLocations,
    includeProcesses,
    includeManufacturers,
});

  console.log("Product status:", products);
```

Which the user has it eternally on "isLoading: true" and the fetchStatus is accused as "paused" for some reason? After logging the status, that's what my user sees:

```
data: undefined
dataUpdatedAt: 0
error: null
errorUpdateCount: 0
errorUpdatedAt: 0
failureCount: 0
failureReason: null
**fetchStatus: "paused"**
**isError: false**
isFetched: false
isFetchedAfterMount: false
isFetching: false
isInitialLoading: false
**isLoading: true**
isLoadingError: false
**isPaused: true**
isPlaceholderData: false
isPreviousData: false
isRefetchError: false
isRefetching: false
isStale: true
isSuccess: false
```

As seen, I'm not pausing the request myself for whatsoever reason.

The only special thing I'm doing is running a middleware for sentry and Clerk authorization.

```
const sentryMiddleware = t.middleware(
  Sentry.Handlers.trpcMiddleware({
    attachRpcInput: true,
  })
);

const enforceUserIsAuthorized = t.middleware(async ({ ctx, next }) => {
  if (!ctx.currentUser)
    throw new TRPCError({
      code: "UNAUTHORIZED",
      message: "You need to be authorized to do this.",
    });

  return next({ ctx: { currentUser: ctx.currentUser } });
});

export const privateProcedure = t.procedure
  .use(sentryMiddleware)
  .use(enforceUserIsAuthorized);
```

**Comment by KATT** [comment]
Might be that their network is disconnected? React query has some smart defaults like not trying to fetch while offline 

https://tanstack.com/query/v4/docs/react/guides/network-mode#network-mode-online

**Reply by russ0133** [reply, marked as answer]
It seems like it was some temporary issue with their browser/computer/network/whatever, however I did make a commit to try and fix it by changing how I handled middlewares, but I doubt this was the fix. Sharing it here for future people who might have same issue.

They were not disconnected from network though as all the other frontend components worked fine, and I was on call with them at the moment. Just the components that relied on trpc queries weren't rendering.

![image](https://github.com/trpc/trpc/assets/26175960/4893612e-41d9-4a5e-8f1d-d312cea94f7e)
